spec.md

```diff
@@ -99,7 +97,8 @@ The `Span` interface must have the following capabilities:
 
 The `Tracer` interface must have the following capabilities:
 
-- Start a `Span` that has no parent, commonly referred to as a *root* `Span` **(py: `start_trace`, go: `StartTrace`)**
+- Start a new child `Span` with a given operation name. With no parent, commonly referred to as a *root* `Span`, no trace attributes exist. With a parent any trace attributes must be passed through into the created child. **(py: `create_span`, go: `CreateSpan`)**
```

Comment:
@michaelsembwever how would I implement a case where I would like to start a new Span _only if_ a previously existing trace context (eg. upstream span) exists?  Is that something that the instrumenter must now put explicitly into their logic?

The reason I ask about this case is that I think it is a common one for sampling-based tracers: you would not want to trigger a sampling decision at every Span, typically only ones at entry points into services.  If the only way to create a Span is the method `create_span` on `Tracer`, the `Tracer` implementation doesn't have much information about whether it ought to consider sampling or not.  This would demand the instrumenter put guards around many calls to `create_span`.

As an example, if `create_span` would potentially engage sampling logic in any instance where a parent Span is not provided:
```
def my_service_handler(req):
    span = create_span()
    do_stuff(span)
    span.finish()

def do_stuff(span)
    if (span)
        span2 = create_span(span)
    # xxx
    if (span)
        span2.finish()
```

As instrumentation within services for things like RPC clients, DB clients, etc grows, the case exhibited in `do_stuff` becomes much more common than the other case.

Therefore I think this is worth supporting by having two span creation methods: one which signals to the tracer that a unit of sample-able work is beginning, and one that does not.  Tracers might choose to implement the two exactly the same under the hood, but I suspect many will treat them differently.  X-Trace-based approaches benefit greatly from this today and I believe Zipkin/Dapper models do as well.